python/lsst/obs/base/cli/opt/instrument.py

```diff
@@ -0,0 +1,36 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import click
+
+__all__ = ["instrument_option"]
+
+
+class instrument_option:  # noqa: N801
+    def __init__(self, required=False, helpMsg=None):
+        self.required = required
+        self.help = helpMsg if helpMsg is not None else "The name of an Instrument subclass."
+
+    def __call__(self, f):
+        return click.option("-i", "--instrument",
```

Comment:
Can you make this use a callback that converts the supplied string to an `Instrument` instance? (I have no idea whether click can support that).

Reply:
yup.

Reply:
...good idea.

Reply:
I think there's a pretty good chance we'll want to reserve `-i` for something else, or at least `pipetask` in `ctrl_mpexec` already uses it to mean something else.

I'm not sure if we should avoid short options entirely until we've seen the big picture and can see what might be in contention, but we should at least be prepared to go through and remove/change some later.

Reply:
actually, because making an instrument requires two args (the instrument name and the repo location) this is not a natural fit.